### PR TITLE
net-wireless/unifi: add system-mongodb useflag

### PR DESCRIPTION
**Pull request by expeditioneer**
Closes: https://bugs.gentoo.org/839342

### CODE REVIEW

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+src_compile() {
+	:;
+}
+
+src_install() {
+	insinto /usr/lib/unifi
+	doins -r bin dl lib webapps
+	if ! use system-mongodb; then
+		doins -r bin
```
Comment (ConiKost) [inline net-wireless/unifi/unifi-7.1.61-r1.ebuild:87]: One liner please ```if ! use system-mongodb && doins -r bin```

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	done
+	dosym ../../../var/log/unifi /usr/lib/unifi/logs
+
+	java-pkg_regjar "${D}"/usr/lib/unifi/lib/*.jar
+	java-pkg_dolauncher unifi --java_args '-Dorg.xerial.snappy.tempdir=/usr/lib/unifi/tmp -Djava.library.path=' --jar ace.jar --pwd '/usr/lib/unifi'
+
+	newinitd "${FILESDIR}"/unifi.initd-r2 unifi
+
+	if use system-mongodb; then
+		systemd_newunit "${FILESDIR}"/unifi.service-system-mongodb-r2 unifi.service
```
Comment (ConiKost) [inline net-wireless/unifi/unifi-7.1.61-r1.ebuild:104]: I would like to see a different name in ```FILESDIR```. I would be happy with ```unifi.mongodb.service```.

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+DESCRIPTION="A Management Controller for Ubiquiti Networks UniFi APs"
+HOMEPAGE="https://www.ubnt.com"
+SRC_URI="https://dl.ui.com/unifi/${PV}${RC_SUFFIX}/UniFi.unix.zip -> ${P}.zip"
+S="${WORKDIR}/UniFi"
+
+KEYWORDS="-* ~amd64 ~arm64"
+LICENSE="Apache-1.0 Apache-2.0 BSD-1 BSD-2 BSD CDDL EPL-1.0 GPL-2 LGPL-2.1 LGPL-3 MIT ubiquiti"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="systemd system-mongodb"
+REQUIRED_USE="system-mongodb? ( systemd )"
```
Comment (ConiKost) [inline net-wireless/unifi/unifi-7.1.61-r1.ebuild:20]: What about OpenRC? Couldn't be provide an updated init script with ```rc_need mongodb``` and ```rc_after mongodb```?

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	virtual/jre:1.8
+"
+
+BDEPEND="app-arch/unzip"
+
+DOCS=( "readme.txt" )
+
+QA_PREBUILT="usr/lib/unifi/lib/native/Linux/x86_64/*.so"
+
+DOC_CONTENTS="
```
Comment (ConiKost) [inline net-wireless/unifi/unifi-7.1.61-r1.ebuild:36]: Please move the content into ```"${FILESDIR}/README.gentoo"```

**Review by ConiKost — CHANGES_REQUESTED** [review]

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild (hunk 1 above)
Comment (expeditioneer) [reply net-wireless/unifi/unifi-7.1.61-r1.ebuild]: DONE

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild (hunk 2 above)
Comment (expeditioneer) [reply net-wireless/unifi/unifi-7.1.61-r1.ebuild]: DONE

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild (hunk 3 above)
Comment (expeditioneer) [reply net-wireless/unifi/unifi-7.1.61-r1.ebuild]: Added OpenRC support

File: net-wireless/unifi/unifi-7.1.61-r1.ebuild (hunk 4 above)
Comment (expeditioneer) [reply net-wireless/unifi/unifi-7.1.61-r1.ebuild]: DONE

**Review by ConiKost — APPROVED** [review]

**Pull request closed without merging**